### PR TITLE
pv: 1.6.20 -> 1.7.24

### PR DESCRIPTION
**Pull request by matthiasbeyer**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 249594` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>appimage-run-tests</li>
  </ul>
</details>
<details>
  <summary>38 packages failed to build:</summary>
  <ul>
    <li>davinci-resolve</li>
    <li>passky-desktop</li>
    <li>pcloud</li>
    <li>plexamp</li>
    <li>polypane</li>
    <li>protonup-qt</li>
    <li>pureref</li>
    <li>radicle-upstream</li>
    <li>rambox</li>
    <li>raven-reader</li>
    <li>remnote</li>
    <li>requestly</li>
    <li>revolt-desktop</li>
    <li>ripcord</li>
    <li>saleae-logic-2</li>
    <li>session-desktop</li>
    <li>sonixd</li>
    <li>soulseekqt</li>
    <li>ssb-patchwork</li>
    <li>station</li>
    <li>steam-rom-manager</li>
    <li>super-productivity</li>
    <li>timeular</li>
    <li>todoist-electron</li>
    <li>trezor-suite</li>
    <li>tusk</li>
    <li>uhk-agent</li>
    <li>uhk-udev-rules</li>
    <li>upscayl</li>
    <li>via</li>
    <li>vial</li>
    <li>wootility</li>
    <li>youtube-music</li>
    <li>ytmdesktop</li>
    <li>zecwallet-lite</li>
    <li>zettlr</li>
    <li>zettlr-beta</li>
    <li>zulip</li>
  </ul>
</details>
<details>
  <summary>64 packages built:</summary>
  <ul>
    <li>Sylk</li>
    <li>altair</li>
    <li>anytype</li>
    <li>appimage-run</li>
    <li>beekeeper-studio</li>
    <li>betterdiscord-installer</li>
    <li>blockbench-electron</li>
    <li>bloomrpc</li>
    <li>bootstrap-studio</li>
    <li>buttercup-desktop</li>
    <li>caprine-bin</li>
    <li>chrysalis</li>
    <li>cider</li>
    <li>codux</li>
    <li>cozy-drive</li>
    <li>crypto-org-wallet</li>
    <li>devdocs-desktop</li>
    <li>electron-mail</li>
    <li>electronplayer</li>
    <li>firefly-desktop</li>
    <li>flexoptix-app</li>
    <li>fluent-reader</li>
    <li>framesh</li>
    <li>freetube</li>
    <li>fspy</li>
    <li>golden-cheetah-bin</li>
    <li>hamsket</li>
    <li>irccloud</li>
    <li>jbrowse</li>
    <li>jetbrains-toolbox</li>
    <li>jitsi-meet-electron</li>
    <li>joplin-desktop</li>
    <li>keet</li>
    <li>lbry</li>
    <li>ldtk</li>
    <li>ledger-live-desktop</li>
    <li>lens</li>
    <li>localsend</li>
    <li>logseq</li>
    <li>losslesscut-bin</li>
    <li>lunar-client</li>
    <li>lunatask</li>
    <li>marktext</li>
    <li>mathpix-snipping-tool</li>
    <li>mendeley</li>
    <li>mobilecoin-wallet</li>
    <li>mockoon</li>
    <li>molotov</li>
    <li>motrix</li>
    <li>museeks</li>
    <li>mycrypto</li>
    <li>neo4j-desktop</li>
    <li>notable</li>
    <li>notesnook</li>
    <li>notion-app-enhanced</li>
    <li>nuclear</li>
    <li>octant-desktop</li>
    <li>openlens</li>
    <li>osu-lazer-bin</li>
    <li>p3x-onenote</li>
    <li>pv</li>
    <li>sanoid</li>
    <li>wal_e</li>
    <li>wal_e.dist</li>
  </ul>
</details>

File: pkgs/tools/misc/pv/default.nix [hunk 1]
```diff
@@ -6,11 +6,11 @@
...
+  version = "1.7.24";
 
   src = fetchurl {
-    url = "https://www.ivarch.com/programs/sources/pv-${version}.tar.bz2";
-    sha256 = "00y6zla8h653sn4axgqz7rr0x79vfwl62a7gn6lzn607zwg9acg8";
+    url = "https://www.ivarch.com/programs/sources/pv-${version}.tar.gz";
+    sha256 = "sha256-O/Q8WAnI1QBm6urqWhFfZQPFejjBUZdbcQqivuhXtl4=";
   };
 
   patches = [
```
Comment (anthonyroussel) [inline pkgs/tools/misc/pv/default.nix:16]: I think this patch is not needed anymore.

Looks like the issue has been fixed with these commits:

* https://codeberg.org/a-j-wood/pv/commit/f13f5e70c1b84cd748cd9e12d55251d78449e1b6
* https://codeberg.org/a-j-wood/pv/commit/7448c7f320d25c43dfd3da8d126a26ec65c33d3e

**Comment by matthiasbeyer** [comment]
Incooporated your suggestions and rebased+squashed! :smile:

**Review by anthonyroussel — APPROVED** [review]

File: pkgs/tools/misc/pv/default.nix [hunk 2]
```diff
@@ -6,26 +6,18 @@
...
   src = fetchurl {
-    url = "https://www.ivarch.com/programs/sources/pv-${version}.tar.bz2";
-    sha256 = "00y6zla8h653sn4axgqz7rr0x79vfwl62a7gn6lzn607zwg9acg8";
+    url = "https://www.ivarch.com/programs/sources/pv-${version}.tar.gz";
+    sha256 = "sha256-O/Q8WAnI1QBm6urqWhFfZQPFejjBUZdbcQqivuhXtl4=";
   };
 
-  patches = [
-    # Fix build on aarch64-darwin using patch from Homebrew
-    (fetchpatch {
```
Comment (anthonyroussel) [inline pkgs/tools/misc/pv/default.nix:18]: Don't forget to remove the unused fetchpath import at the top of the file.

**Review by anthonyroussel — APPROVED** [review]

**Review by anthonyroussel — APPROVED** [review]

File: pkgs/tools/misc/pv/default.nix [hunk 3]
```diff
@@ -1,31 +1,22 @@
...
-  ];
-
   meta = {
-    homepage = "http://www.ivarch.com/programs/pv";
+    homepage = "https://www.ivarch.com/programs/pv.shtml";
     description = "Tool for monitoring the progress of data through a pipeline";
     license = lib.licenses.artistic2;
-    maintainers = with lib.maintainers; [ ];
+    maintainers = with lib.maintainers; [ matthiasbeyer ];
     platforms = with lib.platforms; all;
```
Comment (AndersonTorres) [inline pkgs/tools/misc/pv/default.nix:20]:
```suggestion
    platforms = lib.platforms.all;
```

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**